### PR TITLE
trie: track dirty cache metrics, track clean writes on commit

### PR DESCRIPTION
**Pull request by karalabe**
This is needed to help better compare the snapshotter PR with master.

### CODE REVIEW

File: trie/database.go [hunk 1]
```diff
@@ -372,8 +379,12 @@ func (db *Database) node(hash common.Hash) node {
 	db.lock.RUnlock()
 
 	if dirty != nil {
+		memcacheDirtyHitMeter.Mark(1)
+		memcacheCleanReadMeter.Mark(int64(dirty.size))
```
Comment (holiman) [inline trie/database.go]: Shouldn't it be this?
```suggestion
		memcacheDirtyReadMeter.Mark(int64(dirty.size))
```

File: trie/database.go [hunk 2]
```diff
@@ -408,8 +419,12 @@ func (db *Database) Node(hash common.Hash) ([]byte, error) {
...
```
Comment (holiman) [inline trie/database.go]:
```suggestion
		memcacheDirtyReadMeter.Mark(int64(dirty.size))
```
?

File: trie/database.go (hunk 1 above)
Comment (karalabe) [reply trie/database.go]: Damn, I though I fixed this.

Comment (karalabe) [reply trie/database.go]: It's really weird, I found this issue on Friday and could have swarn I fixed it
apparently maybe I didn't commit

File: trie/database.go (hunk 2 above)
Comment (karalabe) [reply trie/database.go]: Yup, god catch, fixed.

File: trie/database.go (hunk 1 above)
Comment (karalabe) [reply trie/database.go]: Fixed.

**Pull request merged**